### PR TITLE
feat(amplify-appsync-simulator): implement `$utils.rds.toJsonObject`

### PR DESCRIPTION
**Pull request by jariz**
#### Description of changes

This implements [`$utils.rds.toJsonObject`](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-util-reference.html#rds-helpers-in-util-rds) in velocity.

#### Description of how you validated changes
Test was added.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)  
   _N/A_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cjihrig — COMMENTED** [review]
Thanks for the contribution. Left a few small comments.

File: packages/amplify-appsync-simulator/src/velocity/util/rds.ts [hunk 1]
```diff
@@ -1,19 +1,32 @@
+const serializeRdsObject = rdsObject => {
+  return (rdsObject?.sqlStatementResults ?? []).map(statement =>
+    (statement?.records ?? []).map(record => {
+      const result = {};
+      record.forEach((row, index) => {
+        result[statement?.columnMetadata?.[index]?.name] = Object.values(row)?.[0];
```
Comment (katesclau) [inline packages/amplify-appsync-simulator/src/velocity/util/rds.ts:6]: Hey @jariz, I know `toJsonString` also does not consider `null` values, but I think we could also update this in the same PR.

```
result[statement?.columnMetadata?.[index]?.name] = row['isNull'] || row['null'] ? null : Object.values(row)?.[0];
```
^ would account for both `{ "isNull": true, "null": true }`

File: packages/amplify-appsync-simulator/src/velocity/util/rds.ts [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
       return '';
     }
   },
+
+  toJsonObject: rdsString => {
+    try {
+      const rdsObject = JSON.parse(rdsString);
+      return serializeRdsObject(rdsObject);
+    } catch (ex) {
+      return '';
```
[the remark this answers is not in the document]
Comment (jariz) [reply packages/amplify-appsync-simulator/src/velocity/util/rds.ts:29]: I'm going to mark this as resolved. If you have any further suggestions on what to do here, I'd still love to hear it.

**Comment by jariz** [comment]
All feedback processed, can someone take a look at this again? 😄

**Review by katesclau — APPROVED** [review]

**Review by katesclau — APPROVED** [review]

**Review by katesclau — APPROVED** [review]

**Review by pavellazar — COMMENTED** [review]
changes LGTM, but there are merge conflicts, I can approve after updated

**Comment by jariz** [comment]
@lazpavel resolved

**Review by pavellazar — APPROVED** [review]

**Review by sachscode — APPROVED** [review]
LGTM! Thanks a lot for the contribution

**Pull request merged**